internal/provider/space_resource.go

```diff
@@ -67,9 +61,9 @@ func (r *SpaceResource) Schema(ctx context.Context, req resource.SchemaRequest,
 				},
 			},
 			"id": schema.StringAttribute{
-				MarkdownDescription: "Id of the space. Must be globally unique.",
-				Computed:            true,
+				MarkdownDescription: "Id of the space. Must be globally unique. If the provider has a space configured and this field is not provided, the provider space is used.",
```

Comment:
Is there a use case where people will pre-populate the provider with some space id and use it to actually create the space?  I think its probably better for us if we enforce this to be set on the space resource

Reply:
I love this feedback, but I think I am bias since I think that most resources should honor the provider configuration if it is configured. I can see two use cases.

## A provider with a space configured

This will allow users do everything in a single space, and also create new spaces outside the default one, something like:
```hcl
provider "mondoo" {
  space = "hungry-poet-123"
}

resource "mondoo_space" "hungry" {
  name = "A hungry space"
  org_id = "cool-security-organization"
}

resource "mondoo_integration_slack" "hungry" {
  name = "My hungry Slack integration"
  ...
}

resource "mondoo_integration_shodan" "hungry" {
  name = "My hungry Shodan integration"
  ...
}

# Also, if they want a different space, they can do this
resource "mondoo_space" "strong" {
  name    = "A strong space"
  space_id = "strong-new-space-12345"
  org_id   = "cool-security-organization"
}
resource "mondoo_integration_shodan" "strong" {
  space_id = mondoo_space.strong.id
  name = "My strong Shodan integration"
  ...
}

```
## A provider WITHOUT space configured

Here the user needs to specify every single time the space, something like this:
```hcl
provider "mondoo" { }

resource "mondoo_space" "first" {
  name = "First space"
  org_id = "cool-security-organization"
}

resource "mondoo_integration_slack" "first" {
  space_id = mondoo_space.first.id
  name = "My first Slack integration"
  ...
}

resource "mondoo_integration_shodan" "first" {
  space_id = mondoo_space.first.id
  name = "My first Shodan integration"
  ...
}

# And for a new different space, the same
resource "mondoo_space" "second" {
  name    = "Second space"
  org_id   = "cool-security-organization"
}
resource "mondoo_integration_shodan" "second" {
  space_id = mondoo_space.second.id
  name = "My second Shodan integration"
  ...
}
```